.github/workflows/web-publish.yml

```diff
@@ -0,0 +1,56 @@
+# Based on sample workflow for building and deploying a Jekyll site to GitHub Pages
+name: Deploy to GitHub pages with Jekyll
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["issue38-publish_to_web"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+            persist-credentials: false
+      - name: Setup Pages
+        uses: actions/configure-pages@983d7736d9b0ae728b81ab479565c72886d7745b # v5.0.0
+      - name: Build content from yaml
+        run:  cd cmd && go run main.go compile --output ../docs/index.md
```

Comment:
Do you think this step needs [actions/setup-go](https://github.com/actions/setup-go) first?

Reply:
sounds good! TIL go is in the runner :)